### PR TITLE
Logout button will use the logout API

### PR DESCRIPTION
**Pull request by michielderoos**
**Describe the Pull Request**

Logout button was just nuking the cookie instead of also hitting the logout API (which invalidates the JWT)

**Todo**

- [x] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes, checking [accessibility](https://www.a11yproject.com/checklist/) etc.
- [x] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver
- [ ] Add release notes to [unreleased changelog](https://github.com/teamsempo/SempoBlockchain/blob/master/CHANGELOG.md#unreleased)

### CODE REVIEW

**Comment by michielderoos** [comment]
[Fix logout button](https://trello.com/c/Va1fvchy/832-fix-logout-button)

**Review by tristanhcole — CHANGES_REQUESTED** [review]
Code makes sense but I'm getting a `Status Code: 401 UNAUTHORIZED` when I log out and token isn't being added to the table `blacklist_tokens`
Screen Shot 2021-05-12 at 10 03 32 am

File: app/server/api/auth_api.py [hunk 1]
```diff
@@ -480,6 +480,7 @@ def post(self):
             auth_token = auth_header.split(" ")[0]
         else:
             auth_token = ''
+        auth_token = auth_header.split("|")[0]
```
Comment (michielderoos) [inline app/server/api/auth_api.py:483]: Note the BE fix here-- I think the logic here predates our TFA tokens

Comment (tristanhcole) [reply app/server/api/auth_api.py:483]: that makes sense

File: app/client/sagas/authSagas.ts [hunk 2]
```diff
@@ -177,6 +178,13 @@ function* saveOrgId(
 function* watchSaveOrgId() {
   yield takeEvery(LoginActionTypes.UPDATE_ACTIVE_ORG, saveOrgId);
 }
+
+export function* apiLogout() {
+  yield call(logoutAPI);
+  yield call(removeSessionToken);
+  yield call(removeOrgIds);
```
Comment (tristanhcole) [inline app/client/sagas/authSagas.ts:185]: Can you explain the reasoning for why there's a separate `apiLogout` and `logout` function?

**Review by tristanhcole — COMMENTED** [review]
One small question that would be good to have an answer for, otherwise tested and make sense to me

File: app/client/sagas/authSagas.ts (hunk 2 above)
Comment (michielderoos) [reply app/client/sagas/authSagas.ts:185]: Two reasons! One is that we don't necessarily want to blacklist every time we flush the session token (namely when we're refreshing the token, we don't want to invalidate the token we're about to refresh with).   
The other is that we sometimes call `logout` with no token at all, or with an expired token (Like when you go directly to the `/login` page). These cases would just fail and there's no use wasting an API call on them IMO

Comment (tristanhcole) [reply app/client/sagas/authSagas.ts:185]: That makes sense

**Review by tristanhcole — APPROVED** [review]
👍

**Pull request merged**